paasta_tools/cli/cmds/spark_run.py

```diff
@@ -808,7 +841,17 @@ def paasta_spark_run(args):
         return 1
 
     volumes = instance_config.get_volumes(system_paasta_config.get_volumes())
-    app_base_name = get_spark_app_name(args.cmd or instance_config.get_cmd())
+    app_base_name = get_spark_app_name(args.cmd or instance_config.get_cmd(), args.autogen)
+
+    if args.autogen != '-1':
+        pod_template_path = "/nail/tmp/podTemplate.yaml"
+        if args.spark_args:
```

Comment:
I don't think we should append to args.spark_args as these are not the user given args.
It's better to append the these key/values after args.spark_args are parsed. 
Line: https://github.com/Yelp/paasta/pull/3203/files#diff-30f8e9c33224158fab1e77efc4004f7d2344230ea7b4ffbf64b3fca21513cc00R856
`    user_spark_opts = _parse_user_spark_args(args.spark_args)`